api/lib/flight_job_script_api/model_cache.rb

```diff
@@ -45,6 +45,12 @@ def find!(id, **opts)
 
   private
 
+  # NOTE: Technically the 'id' is not globally unique for all models and
+  # depends on the user make the request. This is fine whilst the models are
+  # cached on a per-request basis, as they can only be a single user.
+  #
+  # However this assumption might fail if the cache is persisted across requests.
+  # Consider refactoring.
```

Comment:
Can we have a less worrying note here.  I can't see that there is anything to consider refactoring here. `RequestStore` is as its name suggests a store for a single request.  Given that, the `cache_id` is perfectly reasonable.  Being explicit about the limitation of the cache/cache id is fine, but the suggestion to refactor suggests to me that there is something lacking that needs to be addressed, rather than a considered limitation.  If you see the difference.

Reply:
Updated